### PR TITLE
Add volumetric inlet, revamp comments for Jacobian derivation, and debug Jacoabian for interfaces

### PR DESCRIPTION
**Pull request by hwpang**

### CODE REVIEW

**Review by mjohnson541 — CHANGES_REQUESTED** [review]

File: src/Interface.jl [hunk 1]
```diff
@@ -318,6 +318,43 @@ struct VolumetricFlowRateOutlet{V,F1<:Function} <: AbstractBoundaryInterface
...
+
+    if haskey(conddict,"T")
+        T = conddict["T"]
+    end
+    if haskey(conddict,"P")
+        P = conddict["P"]
+    end
+
+    if haskey(conddict,"Hin")
+        H = conddict["Hin"]
```
[the remark this answers is not in the document]
Comment (hwpang) [reply src/Interface.jl]: I changed Hin to be J/mol to be consistent with the Hin in Inlet, and calculate Hpervolume (J/m^3) to append to the domain.

File: src/Domain.jl [hunk 2]
```diff
@@ -2065,6 +2098,20 @@ end
...
                 @inbounds @fastmath jac[domain.indexes[4],i] -= kLAs[i]/kHs[i]*R*T/V + P/T*ddnidTdt
             end
+        elseif isa(inter,VolumetricFlowRateInlet) && domain == inter.domain
+            # dn/dt = inter.Vin(t)*inter.cs
+            # dT/dt = inter.Vin(t)*(inter.H - dot(Us,ns)/V)/(N*Cvave)
+            # dP/dt = inter.Vin(t)*P/V + P/T*dTdt
+            # d/dni(dT/dt) = inter.Vin(t)*(-Us[i]/N)/(N*Cvave)-dTdt*(dCvavedni/Cvave)
+            # d/dni (dP/dt) = P/T * d/dni(dT/dt) 
+            @fastmath dTdt = (inter.Vin(t)*(inter.H - dot(Us,ns)/V))/(N*Cvave)
+            @simd for i in domain.indexes[1]:domain.indexes[2]
```
[the remark this answers is not in the document]
Comment (hwpang) [reply src/Domain.jl:2251]: Done

File: src/Domain.jl [hunk 3]
```diff
@@ -2485,6 +2550,17 @@ end
...
                 @inbounds @fastmath jac[domain.indexes[4],i] -= kLAs[i]/kHs[i]*R*T/V + P/T*ddnidTdt
             end
+        elseif isa(inter,VolumetricFlowRateInlet) && domain == inter.domain
+            # dn/dt .+= inter.Vin(t)*inter.cs
+            # dT/dt += inter.Vin(t)*(inter.H - dot(Us,ns)/V)/(N*Cvave)
+            # dP/dt += inter.Vin(t)*P/V + dT/dt*P/T
+            # d/dni(dT/dt) = -inter.Vin(t)*(Us[i]/V)/(N*Cvave)
+            # d/dni(dP/dt) = d/dni(dT/dt)*P/T
+            @simd for i in domain.indexes[1]:domain.indexes[2]
+                ddnidTdt = inter.Vin(t)*(Us[i]/V)/(N*Cvave)
```
[the remark this answers is not in the document]
Comment (hwpang) [reply src/Domain.jl]: Fixed

File: src/Domain.jl [hunk 4]
```diff
@@ -2617,6 +2693,25 @@ end
...
+            # d/dV(dT/dt) = inter.Vin(t)*(dot(Hs,ns)/V^2)/(N*Cpave) - dT/dt/(N*Cpave) * d/dV(PV/RT*Cpave)
+            #             = inter.Vin(t)*(dot(Hs,ns)/V^2)/(N*Cpave) - dT/dt/(N*Cpave) * (P/RT*Cpave + PV/RT*dCpavedV)
+            #               Note: Cpave = dot(cpdivR,ns)*R/N = dot(cpdivR,ns)*R*(RT/PV)
+            #               Note: d/dV(Cpave) = dot(cpdivR,ns)*R*(RT/PV)*(-1/V) = -Cpave/V
+            #             = inter.Vin(t)*(dot(Hs,ns)/V^2)/(N*Cpave) - dT/dt/(N*Cpave) * (P/RT*Cpave + PV/RT(-Cpave/V))
+            #             = 0
+            dTdt = inter.Vin(t)*(inter.H - dot(Hs,ns)/V)/(N*Cpave)
+            @simd for i in domain.indexes[1]:domain.indexes[2]
+                @inbounds @fastmath dCpavedni = cpdivR[i]*R/N
+                @inbounds @fastmath jac[domain.indexes[3],i] += inter.Vin(t)*(Hs[i]/V)/(N*Cpave) - dTdt*(dCpavedni/Cpave)
```
[the remark this answers is not in the document]
Comment (hwpang) [reply src/Domain.jl]: Fixed

**Pull request merged**